packages/shared/components/filters/Filter.svelte

```diff
@@ -1,12 +1,13 @@
 <script lang="typescript">
     import { TogglableButton, FilterModal, FilterItem, Modal } from 'shared/components'
-    import type { Filter } from '@core/wallet/interfaces/filter/filter.interface'
+    import type { ActivityFilter, AssetFilter } from '@core/wallet/interfaces'
     import { deepCopy } from '@core/utils'
     import type { Writable } from 'svelte/store'
+    import { ProposalFilter } from '@contexts/governance'
 
-    export let filterStore: Writable<Filter>
+    export let filterStore: Writable<ActivityFilter | AssetFilter | ProposalFilter>
 
-    let filter: Filter = deepCopy($filterStore)
+    let filter: ActivityFilter | AssetFilter | ProposalFilter = deepCopy($filterStore)
```

Comment:
What happened to Filter interface and why not keep it as a type union?

Reply:
i moved all of the general filter stuff into the `utils` module.
So now e.g. the `ActivityFilter` imports things from `utils`.
If i introduce again a generic `Filter` type, i'd need again to import eg. `ActivityFilter`, which already imports `utils`. 

I want to avoid this circular dependency, but i could create a `Filter`-type here in this component